### ISSUE TITLE
Stuff to do for the anniversary update

### ISSUE THREAD

**Issue by Wruczek** [Wruczek/ts-website#36]
More info about ts-website v 2.0 (anniversary update) can be found [here](https://github.com/Wruczek/ts-website/milestone/1).

Plans:
- Design and technologies:
  - Bootstrap 4 / Bootswatch 4
  - jQuery 3
  - [PHPFastCache v6](https://github.com/PHPSocialNetwork/phpfastcache/wiki/%5BV6%5D-Changelog)
    - Set the timezone before loading to avoid PHP errors
- New installation wizard
  - Design and technologies:
    - Simple Bootstrap page based on [`lumen`](https://bootswatch.com/4-alpha/lumen/)
    - [nav wizard](https://github.com/acornejo/bootstrap-nav-wizard) indicating the progress
    - When showing a compatibility list:
      - Success: [`fa-check-circle`](http://fontawesome.io/icon/check-circle/) with `#28B62C`
      - Partial-support: [`fa-minus-circle`](http://fontawesome.io/icon/minus-circle/) with `#FF851B`
      - Fail: [`fa-times-circle`](http://fontawesome.io/icon/times-circle/) with `#FF4136`
  - [ ] Check system compatibility before installation
  - [ ] Make sure that caching system works and have proper permissions to write in folders
    - [ ] If not, then try to change it. If no permissions then ask user to do it; provide a command to run.
  - [ ] Ask for MySQL credentials and check the connection
  - [ ] Install tables
  - [ ] Ask for TeamSpeak credentials and then check the connection and permissions. Check for the flood values.
  - [ ] Save everything to config file
  - [ ] After installation is finished, save a boolean in the new config file. If the config file already exists, do not let the installer to be run again.

**Comment by Wruczek** [comment]
Update: shop has been voted out by group assigner on our Telegram group.

**Comment by bykidi** [comment]
@Wruczek like TS-N's younger lighter ranksystem alternative? Dope!

**Comment by Wruczek** [comment]
@bykidi Not really :D It will be a simple group assigner that allows you to change your own groups.

**Comment by avsaponchik** [comment]
@Wruczek still, like TS-N ranksystem module does :D

but the entire ranksystem is heavy, buggy and demands on server resources (two of my 128tick cs go servers uses less cpu than one ranksystem slowed down to custom 2.0s delay)
![default](https://cloud.githubusercontent.com/assets/1337953/26353167/b52b066a-3fc7-11e7-8ec4-5d65d26c35e6.PNG)
I hope, someday, we'll see a brand new fresh, nice looking ranksystem based on your website.

//edit 2018.02.23
ranksystem got optimized so hard, it's now using like ~10% of cpu time compared to older builds

**Comment by Wruczek** [comment]
@bykidi I didnt known about that module. Well yes, this will be replaced by ts-website :D

**Comment by Najsr** [comment]
As I can see you are about to use Latte templating system. Have you ever thought  making whole site within Nette framework? I think it would be easier to code and for future maintenance.

**Comment by Wruczek** [comment]
I'm very new into MVC and OOP in PHP. So far I didn't planned to use any framework, but lately I've stumbled upon [CodeIgniter](https://codeigniter.com/). I'm currently reading through its documentation and it looks great to the point I think of using it in the project. Its small, lightweight, popular and the most important - easy to learn.

**Comment by Najsr** [comment]
Well, some goes for Nette. It's less known thanks to it's origin.

**Comment by avsaponchik** [comment]
Any updates so far?

**Comment by Wruczek** [comment]
Hi @bykidi! Version 2.0 is about 40% completed, but I lack the time to finish it off due to school and private projects I'm doing with few people. I'm sure It will get finished at some point, when I will get some time to work on it. The project is definitely not dead, and I'm not abandoning it - it's just "paused"

**Comment by avsaponchik** [comment]
@Wruczek good to know. good luck out there!

**Comment by ghost** [comment]
You could put on V2 .. a top online users or top server connections .. It would be nice to my opinion. And you can buy a certain degree directly from the website with coins .. etc .. A shop system Just a proposal: D
Continue further.

**Comment by Wruczek** [comment]
@growndex The top online is a great idea.

**Comment by Wruczek** [comment]
![image](https://user-images.githubusercontent.com/7156344/45248918-96c65480-b317-11e8-8857-38622806c86d.png)
👀
no, it's not dead ;)

**Comment by ghost** [comment]
nice bro'.
When will you launch it?

**Comment by Wruczek** [comment]
I hope for alpha (limited admin panel) in max 2 months

**Comment by ghost** [comment]
I understand: Good luck. I'm anxious to test him <3

**Comment by Wruczek** [comment]
![image](https://user-images.githubusercontent.com/7156344/46217888-85370200-c343-11e8-8523-be077ef74e12.png)
I guess the bans page is complete. Finishing the viewer now!

**Comment by ghost** [comment]
Very nice!
congratulations

**Comment by Wruczek** [comment]
Viewer looks good... and the spacers behave just like they should ;)
Next up: group assigner

![image](https://user-images.githubusercontent.com/7156344/47325898-74d62680-d666-11e8-8b9c-2b92248ae338.png)

![image](https://user-images.githubusercontent.com/7156344/47326211-fb3f3800-d667-11e8-812c-79a236cfd728.png)
you can hover over the user to view more details

**Comment by karpalak-mk** [comment]
perfect..waiting for the A release

**Comment by ghost** [comment]
nice

**Comment by raviatex** [comment]
looks very good!

**Comment by derrobin154** [comment]
@Wruczek Can you post a new link to design "lumen" the link is broken

**Comment by Wruczek** [comment]
@derrobin154 Thanks for info, I've updated the issue.

**Comment by derrobin154** [comment]
When can we expect version 2.0? @Wruczek

**Comment by Wruczek** [comment]
I cant promise much, but I hope to release the beta around Christmas ;)

**Comment by robotrono** [comment]
Will it be possible to upgrade from 1v to 2v?

**Comment by Wruczek** [comment]
Nope, you will need to install and configure everything from scratch

**Comment by Bluscream** [comment]
On 4/15 we can celebrate the second birthday of this issue 🎂

**Comment by Wruczek** [comment]
I should rename it, that's for sure

**Comment by KurisuWong** [comment]
Is it any way to add a new lanuage pack? I am now just edited the en language file.
Btw, I can help you to do translation for Chinese Traditional(zh_HK/zh_TW) and Chinese Simplified(zh_CN)

**Comment by Wruczek** [comment]
To add a new language, you can:
1. Copy the language file and paste in with different name, then translate it
2. Add it to the `include/header.php` file

**Comment by Wruczek** [comment]
Merry Christmas everyone!
The first dev release of TS-website 2 has landed!

- [Code](https://github.com/Wruczek/ts-website)
- [Demo](https://ts.wruczek.tech/)
- [Installation tutorial](https://github.com/Wruczek/ts-website/wiki/%5BEN%5D-Website-Installation)

⚠ **Please keep in mind:** This is a development version of TS-website!
There are important features missing, like Admin Panel. Everything is configured via the database. If you don't feel good about it, please wait until the final release.

Enjoy testing and happy new year!

**Comment by Bluscream** [comment]
ouhh shit damn broooh

**Comment by Wruczek** [comment]
Translation platform is open!
Everyone is invited to help translate ts-website [here](https://wruczek.oneskyapp.com/collaboration/project/325562).

**Comment by alexandre433** [comment]
Suggestion add BBcode? Should be safer than using html for **C**_u_**s**tomization

**Comment by Wruczek** [comment]
@alexandre433 html is ok for now. just don't give the option of creating news to random people if you fear xss

**Comment by uetam** [comment]
I would like possibity to additionaly authorize users via existing mysql database before allowing them to add to specify groups.
Or eventually possibility to user additional oauth authorization before allowing them to add to specify groups.

**Comment by MackKnife** [comment]
still in  work or did u stop working on this project ?

**Comment by toster234** [comment]
> still in work or did u stop working on this project ?

The last commit was only eleven days ago. What a dumb question.
If you want to help, just make PR this is an open source project

**Comment by MackKnife** [comment]
> 
> 
> > still in work or did u stop working on this project ?
> 
> The last commit was only eleven days ago. What a dumb question.
> If you want to help, just make PR this is an open source project

well thanks for you quick reminder that im dumb..  maybe  im new in githuib and i cant see where the last "commit iis listed"  but hey  maybe  ur just more into github and  tell new useres to leave...  Besides  the obv.  thanks for the answer.

**Comment by Wruczek** [comment]
> still in work or did u stop working on this project ?

It's not very active, but I don't consider it dead

PS. you are not dumb :)

**Comment by toster234** [comment]
Nobody said you were stupid. Repeated questions always make you nervous.
Don't you think so?

W dniu sob., 17.10.2020 o 17:26 MackKnife <notifications@github.com>
napisał(a):

**Comment by Paddyboy1991** [comment]
https://prnt.sc/xo7ggr
we see that it is not active ;)
but important functions like anti-spam is important and desirable ;) :D
with antispam i release that page instandly on my community! (with minimum edits^^)
